3.processing_features/scripts/1.sc_cosmicqc.py

```diff
@@ -205,12 +202,62 @@ def do_not_adjust_image_brightness(image: Image.Image) -> Image.Image:
 # In[7]:
 
 
-nuclei_high_int_outliers_cdf.sample(n=5, random_state=0)
+nuclei_high_int_outliers_cdf.sample(n=2, random_state=0)
 
 
 # In[8]:
 
 
+# Create a new column 'qc_status' in filtered_combined_df
+copy_filtered_combined_df = filtered_combined_df.copy()
+copy_filtered_combined_df["qc_status"] = copy_filtered_combined_df.index.map(
+    lambda idx: "Failed" if idx in nuclei_high_int_outliers.index else "Passed"
+)
+
+# Set figure size
+plt.figure(figsize=(6, 6))
+
+# Histogram plot for nuclei high intensity outliers, grouped by qc_status
+sns.histplot(
+    data=copy_filtered_combined_df,
+    x="Nuclei_Intensity_UpperQuartileIntensity_DAPI",
+    hue="qc_status",  # Color bins by qc_status
+    multiple="layer",  # Stack the bars for better comparison
+    bins=30,  # Adjust the number of bins as needed
+    kde=False,  # Disable KDE
+    palette={
+        "Passed": "#006400",
+        "Failed": "#CC0099",
+    },  # Custom colors for Passed and Failed
+    legend=False,
+)
+
+# Customize the plot
+plt.xlabel("Nuclei upper quartile intensity (DAPI)", fontsize=14)
+plt.ylabel("Count", fontsize=14)
+plt.xticks(fontsize=12)
+plt.yticks(fontsize=12)
+plt.legend(
+    title="QC Status",
+    loc="upper right",
+    labels=["Passed", "Failed"],
```

Comment:
I think these colors are flipped

Reply:
Thanks for catching that! I will fix it now!

Reply:
Fixed!